feat: Add Caption Rating Endpoint

3dxp opened the pull request:
Add caption rating endpoint with request
validation.

#59

3dxp committed:
feat: Add Caption Rating

Add caption rating endpoint with request
validation.

#59

3dxp committed:
feat: Update Seed Data

Update seed data to conform to new value
structure.

#59

zxkevin commented on packages/hackmcx-api/src/controllers/captions.controller.js:
```diff
@@ -34,4 +34,44 @@ export async function postCaptions(req, res){
+        res.send({error: message})
+    }
+
+    if (!req.body.rating){
```
if req.body.rating = 0 this will trigger even though 0 is a valid input

zxkevin commented on packages/hackmcx-api/src/controllers/captions.controller.js:
```diff
@@ -34,4 +34,44 @@ export async function postCaptions(req, res){
+        return;
+    }
+    let rating = parseFloat(req.body.rating);
+    if (rating.isNaN){
```
This didn't catch NaN, I think syntax is `isNaN(rating)`

3dxp replied:
`isNaN` has other edge case problems. We really wanted `Number.isNaN(rating)`

zxkevin commented on packages/hackmcx-api/src/routers/v1/posts/captions.router.js:
```diff
@@ -1,7 +1,8 @@
-router.post('/', postCaptions)
\ No newline at end of file
+router.post('/', postCaptions)
+router.post("/:captionId/_rate", rateCaption)
```
issue title has it being _rating

3dxp replied:
I think it should be named `_rate`, I will go update that ticket.

3dxp replied:
https://github.com/COSC481W-2022Winter/capstone-hackmcx/issues/59#issuecomment-1065990141

3dxp committed:
feat: Fix Rating Validation

Fix the validation for determining if a
rating presented in a request is a valid
value.

#59

zxkevin reviewed (approved):
Looks Good!

acampb65 reviewed (approved):
Looks good to me
